Merge from upstream v13

Alchez opened the pull request.

gavindsouza committed:
feat: Hash based comparison migration

gavindsouza committed:
fix: "function"al programming++

pateljannat committed:
feat: discussions-component-for-website

(cherry picked from commit 5164b1627d659dd5ba98abf2f3065edc8b44645e)

pateljannat committed:
fix: discussions template and time

(cherry picked from commit 8aa977c5f02bfc32c3e8c609f51aab4e3c4cb00b)

# Conflicts:
#	frappe/public/js/frappe-web.bundle.js
#	frappe/public/scss/website/index.scss
#	frappe/templates/base.html

pateljannat committed:
fix: sider

(cherry picked from commit cd967b36b8f6a11fcee186ce1ece6be62b7e83e5)

pateljannat committed:
fix: sider

(cherry picked from commit 7649988efcd8c939f54ed380badfb50ca2ce9535)

pateljannat committed:
test: ui

(cherry picked from commit dc73dea31838cc599a0b820c6a88dd6c06caebd9)

# Conflicts:
#	frappe/tests/ui_test_helpers.py

pateljannat committed:
fix: sider

(cherry picked from commit 6f64822a55dfb852181ed6abb6691bf691dfe2c6)

pateljannat committed:
fix: increased wait time

(cherry picked from commit 4138d876e2175f8e399f4405d260157b65a47c5f)

pateljannat committed:
fix: removed unnecessary redirect

(cherry picked from commit afdf5611401dcc30b0750d4582fd262519915692)

pateljannat committed:
fix: web page and load time

(cherry picked from commit d49ce24e097cdc68cf8a7a7fb77ffe9269c6b040)

pateljannat committed:
fix: tests

(cherry picked from commit bc5fcce14f060616b5f05912f1e0e326fda10594)

saxenabhishek committed:
fix: missing migration hash column

pateljannat committed:
fix: reply seperator

(cherry picked from commit 367099f1faf0e9dd482b024dcb0c6bc59473f368)

pateljannat committed:
fix: label for web page in template

(cherry picked from commit bd184ccec35addf95d61e2c5ef6aacc17b58af02)

pateljannat committed:
fix: scroll, translations, naming series, placeholder and variable definition

(cherry picked from commit 47fab75bcd68581e82dfba88e00a3d6f4a2151f4)

saxenabhishek committed:
fix: doctype doc hash creation

pateljannat committed:
style: indentation, etc

(cherry picked from commit fd57185bc9376d1dc7ee0c2656829e985c65a3e3)

# Conflicts:
#	frappe/public/scss/website/index.scss
#	frappe/tests/ui_test_helpers.py

saxenabhishek committed:
style: misc changes

gavindsouza committed:
style: Black-ish queries

* feat(minor): Added DocType to frappe.query_builder namespace

saxenabhishek committed:
fix: missing logical cases

- handle first time imports
- update hash and timestamps

saxenabhishek committed:
docs: add docstring for import_file_by_path

vrslev committed:
fix(grid): Make user-wise grid customization translatable

(cherry picked from commit b8470ba7f4806b23d318fa479980b02b8e450b63)

Commit:
feat: allow more print page size options

Commit:
fix: requested changes

ci2014 committed:
Update interaction.js

Add assigned_by to interaction_values for ToDos to avoid PermissionError

(cherry picked from commit 92070cb52a01034c2dda311aba9c1d8903112831)

surajshetty3416 committed:
style: Fix formatting issue

(cherry picked from commit 9bce65b1d7ac20ea43e34d297cb6a3760fecb7e5)

shariquerik committed:
build: Upgrade jQuery 3.5.0 to 3.6.0

(cherry picked from commit 192a975d7fbb07a4b1613d5797b7ce5553d4aab3)

# Conflicts:
#	package.json
#	yarn.lock

sumaiya2908 committed:
fix(Signup Page): Last Step Page style  (#14853)

* fix: template not found

* refactor: scomplete signup page layout

* refactor: Simplify code

Co-authored-by: Summayya <frappe@Summayyas-MacBook-Air.local>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
(cherry picked from commit 88550c1672949c77f6ca29331f76247015dceb74)

shariquerik committed:
fix: resolved conflicts in package.json

shariquerik committed:
fix: resolved conflict in yarn.lock

saxenabhishek committed:
fix: remove unused args

gavindsouza committed:
fix: Update Installed Applications on remove_app

Removing an app from a site should sync the Installed Applications
doctype

(cherry picked from commit 2e28ee5b25ad407d178b19b7efd1a2943603f073)

shariquerik committed:
test: Cypress flaky test (#14565)

gavindsouza committed:
fix(cli): Add new line after each site's migrate

(cherry picked from commit ef39843f6ba8b24912fdb41a62df186dbb39bd15)

shariquerik committed:
test: minor fix

shariquerik committed:
Merge branch 'version-13-hotfix' into check-ui-test-bp

Commit:
fix: pparent doctype freeze

(cherry picked from commit 7f43e102a4f866138c7683039faea31570bde28f)

shariquerik committed:
Merge branch 'version-13-hotfix' into check-ui-test-bp

mergify[bot] committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-14956

surajshetty3416 committed:
Merge pull request #14994 from frappe/mergify/bp/version-13-hotfix/pr-14932

mergify[bot] committed:
Merge pull request #14964 from frappe/mergify/bp/version-13-hotfix/pr-14956

build: Upgrade jQuery v3.5.0 to v3.6.0 (backport #14956)

shariquerik committed:
fix: Update check all checkbox based on checked rows in List View bp v13 (#14975)

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

mergify[bot] committed:
Merge branch 'version-13-hotfix' into check-ui-test-bp

shariquerik committed:
fix(patch): Ignore route conflict validation for Custom to Client Script Rename (#14953)

shariquerik committed:
test: updating timeline.js

sagarvora committed:
fix: optimise `mark_email_as_seen`

(cherry picked from commit 69c87e5caec236ed2b09f52dec73c5c6a2426f9d)

sagarvora committed:
fix: enforce GET method

(cherry picked from commit a80cf47426947651315151cc989211eae7ba23bf)

# Conflicts:
#	frappe/core/doctype/communication/email.py

mergify[bot] committed:
feat(ux): option to disable EPS notifications (bp #14992) 

(cherry picked from commit 3be669ed69869766ca339cefd04f23123384c3e4)

Co-authored-by: Ankush Menat <ankush@frappe.io>

sagarvora committed:
fix: move commit call to `mark_email_as_seen`

(cherry picked from commit 7b7f74ce23c2d9af7d5d6ee01e209385778aa51c)

sagarvora committed:
fix: slightly better naming

(cherry picked from commit cb97b49c78ef9df61eb683d09d262651a738b392)

mergify[bot] committed:
Merge pull request #14996 from frappe/custom-to-client-script-patch-bp

fix(patch): Ignore route conflict validation for Custom to Client Script Rename bp v13

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into check-ui-test-bp

shariquerik committed:
fix: only allow group_by for list view data

(cherry picked from commit bd61ae4675e96be3eb8e7e5bbbbfc531b673b74d)

shariquerik committed:
fix: moved report view related code to report_view.js

(cherry picked from commit 26a49d5f148d08cafbd18e23a322283ff6b74241)

surajshetty3416 committed:
Merge pull request #15009 from frappe/mergify/bp/version-13-hotfix/pr-15008

mergify[bot] committed:
test: print view should not show warning/errors (bp #14972)

(cherry picked from commit 3ee9c0492a0c91663e191e8e4310a477f6f22aa7)

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
test: Cypress test fix for report_view (#14997)

shariquerik committed:
feat: Editable grid/table pagination (#14935)

shariquerik committed:
fix: Only show report when there is data(removed from hide_loading_screen method)

(cherry picked from commit 452146e5640ff2a974442555377d68ad2324930b)

ankush committed:
fix: change subject data type from Data to Text

Subject can be longer than 140 characters. No real reason to restrict it
to arbirary length.

PS: This is done because events are created from other doctypes
automatically and longer subject lines causes a failure where user can't
do much to recover from it.

(cherry picked from commit ed312fc22e86fca80502baceb5ab9722c820355e)

mergify[bot] committed:
Merge pull request #15018 from frappe/mergify/bp/version-13-hotfix/pr-15016

fix: Only show report when there is data (backport #15016)

ankush committed:
test: fix expected failing test

(cherry picked from commit abebd29ef4078737d362986d4b77eaa4c6045887)

mergify[bot] committed:
Merge pull request #15011 from frappe/report-view-ui-test-bp

test: Cypress test fix for report_view bp v13

mergify[bot] committed:
Merge branch 'version-13-hotfix' into check-ui-test-bp

shariquerik committed:
test: fixed failing report_view test

shariquerik committed:
feat: Icon Picker (#13676)

shariquerik committed:
fix: Exporting from report builder not exporting all rows (#15020)

surajshetty3416 committed:
Merge pull request #15026 from frappe/report-view-export-excel-bp

surajshetty3416 committed:
Merge pull request #15030 from frappe/mergify/bp/version-13-hotfix/pr-15027

surajshetty3416 committed:
Merge pull request #15032 from frappe/mergify/bp/version-13-hotfix/pr-14981

surajshetty3416 committed:
Merge pull request #15025 from frappe/icon-picker-bp

surajshetty3416 committed:
Merge pull request #14976 from frappe/check-ui-test-bp

mergify[bot] committed:
Merge branch 'version-13-hotfix' into editable-grid-pagination-bp

surajshetty3416 committed:
Merge pull request #15012 from frappe/editable-grid-pagination-bp

surajshetty3416 committed:
Merge pull request #15035 from frappe/mergify/bp/version-13-hotfix/pr-15017

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
Merge pull request #15005 from frappe/mergify/bp/version-13-hotfix/pr-15001

sagarvora committed:
fix: set `Script Manager` as a standard role

(cherry picked from commit 13c75c805f6559ae467d78b3ebbeada271ba80e5)

Commit:
refactor: 404 page

(cherry picked from commit 99abb51fb68d7da8d3006a037a436aa5e2baa6a7)

surajshetty3416 committed:
Merge pull request #15042 from frappe/mergify/bp/version-13-hotfix/pr-15037

shariquerik committed:
fix: Give select permission to 'All' for workflow state (#15044)

(cherry picked from commit 57614e9a6ee366d90fb48c66e7f808c785d113d0)

surajshetty3416 committed:
Merge pull request #15045 from frappe/mergify/bp/version-13-hotfix/pr-15044

Commit:
refactor: make user facing strings translatable

(cherry picked from commit 6612bb2667064c0d652ab9ca65146909b618909b)

surajshetty3416 committed:
fix: Remove edit icon from quick entry to avoid browser crash

(cherry picked from commit fca5f86a08af426305c1d1dabfe7b7858d87b3a7)

surajshetty3416 committed:
Merge pull request #15050 from frappe/mergify/bp/version-13-hotfix/pr-15049

surajshetty3416 committed:
style: Fix formatting

(cherry picked from commit cb36bfaf045d7f4d18e85df07963567ec9074788)

surajshetty3416 committed:
Merge pull request #15052 from frappe/mergify/bp/version-13-hotfix/pr-15039

barredterra committed:
fix: "send open in web" only if enabled

(cherry picked from commit 1c410de6f5a806b4c50710d86694c557a211cedc)

saxenabhishek committed:
fix: Don't export migration_hash column in std json

Co-authored-by: gavindsouza <gavin18d@gmail.com>

Fixes #14563
backport of 46a8e20ac20d

mergify[bot] committed:
Merge pull request #14986 from saxenabhishek/aks-feat-hash_migration_backport

feat: Hash based comparison migration (backport #14246)

surajshetty3416 committed:
Merge pull request #15061 from frappe/mergify/bp/version-13-hotfix/pr-15056

surajshetty3416 committed:
Merge pull request #15062 from frappe/mergify/bp/version-13-hotfix/pr-14573

shariquerik committed:
fix: Remove chart from report when saved without chart

(cherry picked from commit 06aa8a10adb6dc05c91f2bc7dd589d1ea400b57d)

surajshetty3416 committed:
Merge pull request #15065 from Alchez/v13-granular-printing

surajshetty3416 committed:
Merge pull request #15068 from frappe/mergify/bp/version-13-hotfix/pr-15064

ankush committed:
fix: delete prepared reports on deleting reports

Co-authored-by: Dany Roberts <danyrt@wahni.com>
(cherry picked from commit bacfdaaf7c972dd8b004a48696cfd4f154057c63)

barredterra committed:
fix: navbar settings validation

(cherry picked from commit c30f20c064a1c8531bad4f59ba1069f0e84ccf14)

surajshetty3416 committed:
Merge pull request #15077 from frappe/mergify/bp/version-13-hotfix/pr-15073

surajshetty3416 committed:
Merge pull request #15078 from frappe/mergify/bp/version-13-hotfix/pr-15069

gavindsouza committed:
fix: Don't pass kwargs that make_access_log can't handle

When people try to export reports, sometimes a cmd kwarg is passed
across the three wrapped functions right to make_access_log. This
happens because of the order and way we had to wrap the write_only,
retry and whitelist.

(cherry picked from commit ae9731467ac459b73d685d745f3bb2220384cf39)

# Conflicts:
#	frappe/core/doctype/access_log/access_log.py

mergify[bot] committed:
Merge pull request #15085 from frappe/mergify/bp/version-13-hotfix/pr-15084

fix: Don't pass kwargs that make_access_log can't handle (backport #15084)

surajshetty3416 committed:
Merge pull request #15088 from frappe/mergify/bp/version-13-hotfix/pr-14853

surajshetty3416 committed:
fix: Remove unnecessary style file for note

(cherry picked from commit 48111964ad67de96ce30b961ddcd6357d73f74a0)

gavindsouza committed:
refactor: get_mapping_module doesn't need to access to instance

(cherry picked from commit 5a6d7ee191149291dc2b84391ca8fddeb9543388)

# Conflicts:
#	frappe/data_migration/doctype/data_migration_plan/data_migration_plan.py

gavindsouza committed:
Merge pull request #15096 from frappe/mergify/bp/version-13-hotfix/pr-15095

refactor: get_mapping_module doesn't need to access to instance (backport #15095)

mergify[bot] committed:
fix: Image link render as text in print format (copy #14942) (backport #15091) (#15098)

Co-authored-by: Sagar Sharma <sagarsharma.s312@gmail.com>

surajshetty3416 committed:
fix: Selection color in dark mode

(cherry picked from commit 58ca570f67e47fc3704a5932a14a773e6c87b832)

surajshetty3416 committed:
Merge pull request #15100 from frappe/mergify/bp/version-13-hotfix/pr-15099

bhavesh95863 committed:
fix: remove duplicate parent when child item option

ankush committed:
feat(REST): OR filters in REST API

(cherry picked from commit b1d0e574a0947fdb4133763f63f51677dc9c43ec)

ankush committed:
fix: allow cancelling by PUT docstatus=2

ankush committed:
fix: return self after submit/cancel

`Document.save` returns self but `submit` and `cancel` don't.

change:
For sake of consistency and better support for `run_method` via REST API, return the document.

ref: https://github.com/frappe/frappe/issues/14869
(cherry picked from commit 88c3d92662d4cf05eb1fe4aa1815fdb0f90d53a9)

# Conflicts:
#	frappe/model/document.py

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
Merge pull request #15109 from frappe/mergify/bp/version-13-hotfix/pr-15103

surajshetty3416 committed:
Merge pull request #15107 from frappe/mergify/bp/version-13-hotfix/pr-15105

mergify[bot] committed:
Merge pull request #15101 from bhavesh95863/bhavesh-fix

fix: remove duplicate parent when child item option selected

rohitwaghchaure committed:
chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

mergify[bot] committed:
fix: Use separate API to insert route history (backport #15114) (#15118)

Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>

rohitwaghchaure committed:
Merge pull request #15119 from rohitwaghchaure/merge-hotfix-to-pre-release-for-13-16

chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

mergify[bot] committed:
refactor: Deferred insert for route history (backport #15120) (#15121)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

mergify[bot] committed:
feat: Add an option to disable notification (backport #15102) (#15123)

Co-authored-by: hrwx <himanshuwarekar@yahoo.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

ankush committed:
fix: update EPS records when renaming doc

bhavesh95863 committed:
fix: scrolling issues after minimize global search

mergify[bot] committed:
fix: Reset `lft` & `rgt` in copy doc (backport #15128) (#15129)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

mergify[bot] committed:
fix: Reset `lft` & `rgt` in copy doc (backport #15128) (#15130)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

ankush committed:
fix: reload energy point doctypes in before v13 patches

mergify[bot] committed:
Merge pull request #15126 from ankush/rest_cancel

fix: allow cancelling by PUT docstatus=2 (v13)

mergify[bot] committed:
Merge pull request #15127 from bhavesh95863/fix-minimizepopup

fix: scrolling issues after minimize global search

bhavesh95863 committed:
fix: multiple time global search minimize screen freeze (#15133)

* fix: multiple time global search minimize screen freeze

* chore: format issue

* style: Add missing semicolons

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

oimyounis committed:
fix: add xcall to Frappe's web bundle

Added frappe.xcall to the web bundle to fix 'frappe.xcall is not a function' exception that occurs in Web Form's child tables when choosing a value for a Link field.

mergify[bot] committed:
Merge pull request #15146 from oimyounis/version-13-hotfix

fix: Add `xcall` to Frappe's web bundle

mergify[bot] committed:
fix: Add fallback option for time format when system defaults are not set (backport #15150) (#15153)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

mergify[bot] committed:
chore: better message for system updates (backport #15154) (#15155)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: show "Button" field in user grid config

(cherry picked from commit 66321b075ef06dd73f4d6244f7fa1e99e940d065)

ankush committed:
fix: show "Button" field in user grid config

(cherry picked from commit 66321b075ef06dd73f4d6244f7fa1e99e940d065)

ankush committed:
feat: reset user grid config to default

(cherry picked from commit e0fadef11baf91ac7647538c13e56dd2e0b63dde)

ankush committed:
feat: reset user grid config to default

(cherry picked from commit e0fadef11baf91ac7647538c13e56dd2e0b63dde)

ankush committed:
Merge pull request #15164 from frappe/mergify/bp/version-13-pre-release/pr-15162

feat: reset user-wise grid config to default (backport #15162)

ankush committed:
Merge pull request #15163 from frappe/mergify/bp/version-13-hotfix/pr-15162

feat: reset user-wise grid config to default (backport #15162)

codescientist703 committed:
chore: remove chat module

surajshetty3416 committed:
ci: Use node version 14 to avoid node-sass failure in patch testing build (#15176)

(cherry picked from commit 8a84ae4f0812d2dc5700b7e7011fa96141be204f)

surajshetty3416 committed:
ci: Use node version 14 to avoid node-sass failure in patch testing build (#15176)

(cherry picked from commit 8a84ae4f0812d2dc5700b7e7011fa96141be204f)

codescientist703 committed:
fix: fixed module name

nextchamp-saqib committed:
fix: IndexError while handling sql timeout error

(cherry picked from commit 6f7d030e82ae3b2664f56a6fd9989cfbfc1f1648)

mergify[bot] committed:
refactor: add separate padding for top and bottom in the complete signup page. (backport #15178) (#15181)

Co-authored-by: Summayya <frappe@Summayyas-MacBook-Air.local>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
Co-authored-by: Summayya Hashmani <58825865+sumaiya2908@users.noreply.github.com>

ankush committed:
Merge pull request #15184 from frappe/mergify/bp/version-13-hotfix/pr-15176

ci: Use node version 14 to avoid node-sass failure in patch testing build (backport #15176)

ankush committed:
Merge pull request #15183 from frappe/mergify/bp/version-13-hotfix/pr-15179

fix: IndexError while handling sql timeout error (backport #15179)

mergify[bot] committed:
fix: cannot uninstall app with virtual doctype (backport #15136) (#15185)

Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>
Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: Delete Event Producer Last Update on trash event of Event Producer (backport #15174) (#15187)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

mergify[bot] committed:
Fix: low space on right and left of login (#15178) (#15190)

Co-authored-by: Summayya <frappe@Summayyas-MacBook-Air.local>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
(cherry picked from commit db951d2369d8ca3a13f9f5667706b4a002185983)

Co-authored-by: Summayya Hashmani <58825865+sumaiya2908@users.noreply.github.com>

rohitwaghchaure committed:
chore: change log for v13.16.0

rohitwaghchaure committed:
Merge pull request #15193 from rohitwaghchaure/change-log-for-v13-16

chore: change log for v13.16.0

rohitwaghchaure committed:
bumped to version 13.16.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

mergify[bot] committed:
fix: Clarified docstatus transition exceptions (#15194) (#15197)

* [fix] Clarified docstatus transition exceptions

Exceptions issued by the document.py `check_docstatus_transition` method are potentially very misleading. In cases where an invalid docstatus is used, users receive an confusing exception stating "Cannot change docstatus from 0 to 2" or "Cannot change docstatus from 1 to 0".

This PR adds an additional exception message when an invalid docstatus is used.

* fix: Clarified docstatus transition exceptions

Added additional clarifications to exception messages

(cherry picked from commit 8ead1d9c487b5f7f3e86016b0404a700cc11ba8c)

Co-authored-by: this-gavagai <this-gavagai@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

ghost commented:
### Build System

* Upgrade jQuery 3.5.0 to 3.6.0 ([f8864f7](https://github.com/ParsimonyGit/frappe/commit/f8864f7058d159c6b7c090320f2161d40be5a241))


### Bug Fixes

* resolved conflicts in package.json ([cbb54a0](https://github.com/ParsimonyGit/frappe/commit/cbb54a0bca9b5a9b1b33622d49bd0af82cc2ebe1))
* resolved conflict in yarn.lock ([89be308](https://github.com/ParsimonyGit/frappe/commit/89be30813080dd2b9d42b7e64a629dffc7b38809))
* "function"al programming++ ([0768b60](https://github.com/ParsimonyGit/frappe/commit/0768b60728ff589aa7e5a740ff58aca61483e93d))
* missing migration hash column ([82e1bff](https://github.com/ParsimonyGit/frappe/commit/82e1bfff134d8dcdb02c1d37af9a4000c4c8856e))
* doctype doc hash creation ([1250beb](https://github.com/ParsimonyGit/frappe/commit/1250bebdeb4f1ccdadcc67010c4351c69af68588))
* missing logical cases ([59db38f](https://github.com/ParsimonyGit/frappe/commit/59db38ff9aa778ea42ef13bf0437c700eca21804))
* remove unused args ([bc0c318](https://github.com/ParsimonyGit/frappe/commit/bc0c3182ea6054e11533965ccd550e570143201d))
* Update check all checkbox based on checked rows in List View bp v13 (#14975) ([a04311d](https://github.com/ParsimonyGit/frappe/commit/a04311d86182a85c957259d69ca8cb7c8829b9c1))
*  **patch:** Ignore route conflict validation for Custom to Client Script Rename (#14953) ([5bc1d4a](https://github.com/ParsimonyGit/frappe/commit/5bc1d4a88f191c05dfc1fc205d560f0c00099001))
* optimise `mark_email_as_seen` ([d66728c](https://github.com/ParsimonyGit/frappe/commit/d66728cf490f3295769f73420aee9926789268b1))
* enforce GET method ([d9860fa](https://github.com/ParsimonyGit/frappe/commit/d9860fa3b3c6df4a0dc21f2f07b149b8c0c08f2b))


### Features

* Hash based comparison migration ([9ef09bd](https://github.com/ParsimonyGit/frappe/commit/9ef09bd9f5f9f251b6bee199068ea99d7c9b0449))
*  **ux:** option to disable EPS notifications (bp #14992)  ([58f9dbd](https://github.com/ParsimonyGit/frappe/commit/58f9dbd1ad21c8850331edbdb293bbe9a853fd19))


### Styles

* misc changes ([3cee3ce](https://github.com/ParsimonyGit/frappe/commit/3cee3ce5c3eac3df5db4930fcc2d98875da4173a))
* Black-ish queries ([37f833e](https://github.com/ParsimonyGit/frappe/commit/37f833ec5f49309f942558fe1738ede2ca55ce44))
* Fix formatting issue ([672ed04](https://github.com/ParsimonyGit/frappe/commit/672ed04cb181a769c0c4576ea75c87341c42703d))


### Documentation

* add docstring for import_file_by_path ([4706f1f](https://github.com/ParsimonyGit/frappe/commit/4706f1fbd1c9fc04bed18479b545af71c868d2fb))


### Tests

* Cypress flaky test (#14565) ([ce1229a](https://github.com/ParsimonyGit/frappe/commit/ce1229aac43e58a10c4ff51089eaee851cc43255))
* minor fix ([83d8b15](https://github.com/ParsimonyGit/frappe/commit/83d8b152b620f7c097015438f9f2ade1b0452dcc))
* updating timeline.js ([d9eecd2](https://github.com/ParsimonyGit/frappe/commit/d9eecd21f0c400f7fe19f11aaa188f270a62ee1d))


### Contributors

<a href="https://github.com/shariquerik">shariquerik</a>, <a href="https://github.com/gavindsouza">gavindsouza</a>, <a href="https://github.com/saxenabhishek">saxenabhishek</a>, <a href="https://github.com/mergify">mergify[bot]</a>, <a href="https://github.com/ci2014">ci2014</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/sagarvora">sagarvora</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into remove-chat

pateljannat committed:
fix: conflict in scss

pateljannat committed:
fix: conflicts in template scripts

pateljannat committed:
fix: conflicts in ui test helper

pateljannat committed:
fix: removed include_script from base.html

surajshetty3416 committed:
Merge pull request #15206 from frappe/mergify/bp/version-13-hotfix/pr-14226

mergify[bot] committed:
Merge pull request #15177 from codescientist703/remove-chat

chore: Remove chat module (Backport)

mergify[bot] committed:
fix: ReDoS bug: GHSL-2021-122 (backport #15201) (#15216)

Co-authored-by: Kevin Backhouse <kevinbackhouse@github.com>

ankush committed:
fix: missing content in communication composer

`message` is used as args in some places and `txt` in another.
Supporting both since there's no clear interface to this arguments for
this component.

(cherry picked from commit f589342c7061f6eefca30534676712fd0995b1f5)

ankush committed:
feat: copy error to clipboard

Currently error can be reported if email is setup but other than that
only way to find error info is use browser's console.

This change adds a secondary button to copy error info to clipboard.

(cherry picked from commit 5b6012d0d3944153bb3bb8156c21482a1a8583da)

ankush committed:
fix: copy to clipboard strips off newlines

input element isn't suitable for this hack, switch to textarea for
supporting newlines in text to be copied.

(cherry picked from commit ecdb8932864ced64c9969fe351b492cabdfd10da)

ankush committed:
refactor: use navigator api for clipboard

(cherry picked from commit f3cbaf9df782b16558d9ea7a6811c8ae71c84602)

ankush committed:
fix: disable data import legacy on v13 (#15232)

kennethsequeira committed:
fix: valid parent doctype error (#15233)

Co-authored by: Ankush Menat <me@ankush.dev>

(cherry picked from commit bac62ea014a01d013f8abf502a3ce0481351f84f)

mergify[bot] committed:
Merge pull request #15234 from frappe/mergify/bp/version-13-hotfix/pr-15233

fix: valid parent doctype error (backport #15233)

rohitwaghchaure committed:
fix: record name creating based on naming seires and not as per the use set

(cherry picked from commit 7118fb2e815e6f8567b81e7a0017b2129c9358d1)

rohitwaghchaure committed:
Merge pull request #15241 from frappe/mergify/bp/version-13-hotfix/pr-15238

fix: record name creating based on naming series and not as per the user input (backport #15238)

nemesis189 committed:
fix: Sales invoice cancellation error due to get_list (#15243)

* fix: Sales invoice cancellation error due to get_list

* fix: missed out one `get_list`

(cherry picked from commit 2b8328715e1c084a94c9ddcdccc6a12c8551eebf)

mergify[bot] committed:
fix: update modified of Singles (#15251)

(cherry picked from commit 2ab030c4d813decdc19ecf43bba6601e34a7d721)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: update modified of Singles (#15252)

(cherry picked from commit 2ab030c4d813decdc19ecf43bba6601e34a7d721)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
Merge pull request #15254 from frappe/mergify/bp/version-13-pre-release/pr-15176

ci: Use node version 14 to avoid node-sass failure in patch testing build (backport #15176)

phot0n committed:
fix: not able to go back from user profile page

credit: @ankush
(cherry picked from commit 41bb12a2065a7065fa5fdb63f887f3fa64f95b42)

surajshetty3416 committed:
Merge pull request #15245 from frappe/mergify/bp/version-13-hotfix/pr-15243

shridarpatil committed:
fix: Make query Postgres compatible (#15240)

* fix: updated query compatibale with postgres

* fix: updated query compatible with postgres

(cherry picked from commit 6acbfd5e7466141b134db6091e33d17d74e136e6)

surajshetty3416 committed:
Merge pull request #15266 from frappe/mergify/bp/version-13-hotfix/pr-15240

mergify[bot] committed:
fix: Make query Postgres compatible (backport #15239) (#15267)

Co-authored-by: Shridhar Patil <shridhar.p@zerodha.com>

ankush committed:
fix: remove bench path from traceback

This is mostly useless for debugging and clutters the error message.

(cherry picked from commit 38cd12a95b33fae974e4227e62babfbf3f70e998)

ankush committed:
fix: change route when browsing someone else's profile

(cherry picked from commit faa8c371fb89534777beb37f34317ae33d69c2de)

ankush committed:
fix: leaderboard link in sidebar

(cherry picked from commit a7a05701780f242ff3b562ccae9561e81c87f381)

ankush committed:
fix: remove slash prefix

(cherry picked from commit 00d38d568c0f5f82cca908a08421b1d4d32906b5)

surajshetty3416 committed:
Merge pull request #15269 from frappe/mergify/bp/version-13-hotfix/pr-15265

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
Merge pull request #15270 from frappe/mergify/bp/version-13-hotfix/pr-15268

surajshetty3416 committed:
Merge pull request #15235 from frappe/mergify/bp/version-13-hotfix/pr-15225

vrslev committed:
fix(deps): Drop svg-sprite to fix linux/arm64 build

mergify[bot] committed:
Merge pull request #15275 from vrslev/fix-arm64-build-2

fix(deps): Remove svg-sprite to fix linux/arm64 build

hasnain2808 committed:
revert: set amended docname to original docname-1

Making use_original_name_for_amended_document in system setting to be readonly. This is done to make sure others cannot turn on the setting

(cherry picked from commit 1911b59d3a0ed51f6410b36832f3ea2ca81694ee)

rohitwaghchaure committed:
chore: merge branch 'version-13-hotfix' into 'version-13-pre-release'

barredterra committed:
fix: import doc / fixtures

(cherry picked from commit c92e6ab85d5cbe06713e7ad23c49698c9be3c99c)

rohitwaghchaure committed:
Merge pull request #15286 from rohitwaghchaure/merge-hotfix-to-pre-release-for-13-17

chore: merge branch 'version-13-hotfix' into 'version-13-pre-release'

surajshetty3416 committed:
Merge pull request #15291 from frappe/mergify/bp/version-13-pre-release/pr-15278

surajshetty3416 committed:
Merge pull request #15297 from frappe/mergify/bp/version-13-pre-release/pr-15288

surajshetty3416 committed:
fix: Move barcode style to scss file

So that it applies the style gets applied only in form view and not in printview

(cherry picked from commit a74ce1dba7b61b763ce55d00e76402499155401b)

surajshetty3416 committed:
Merge pull request #15308 from frappe/mergify/bp/version-13-pre-release/pr-15307

pateljannat committed:
fix: datetime in build

(cherry picked from commit ab811415c232117e96d8643b0a09ceb192dbee5a)

pateljannat committed:
fix: using format_datetime instead of prettydate

(cherry picked from commit 74801b05e98ee55b8982907aa478d9337897a511)

pateljannat committed:
fix: removed unnecessary data attributes and class

(cherry picked from commit 850799a33982dc23a88a32db2a466a5103e600f9)

surajshetty3416 committed:
Merge pull request #15316 from frappe/mergify/bp/version-13-pre-release/pr-15314

ankush committed:
fix: set link field to `undefined` instead of empty string

This breaks other code where undefined values are removed, technically
invalid value for link field => unset link field so it should be
undefined

(cherry picked from commit b633f31bf4b3e70c71adad71fba5838fbba0d7de)

surajshetty3416 committed:
refactor:  Use null instead of undefined

(cherry picked from commit 1e3d58c14f614ecac1eb3e5511a13c7ccae12038)

surajshetty3416 committed:
Merge pull request #15327 from frappe/mergify/bp/version-13-pre-release/pr-15320

gavindsouza committed:
fix(ui): Rendering forms created by disabled users

frappe.utils.capitalize doesn't exist anymore (has been very long). And it didn't do that
good of a job for setting fullnames anyway.

(cherry picked from commit f8356141a91c9e11e48da230f369bf80ae1ab1af)

mergify[bot] committed:
fix: Barcode rendering in server side printing (backport #15383) (#15385)

Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>
Co-authored-by: gavin <gavin18d@gmail.com>

rohitwaghchaure committed:
chore: change log for v13.17.0

rohitwaghchaure committed:
Merge pull request #15397 from rohitwaghchaure/change-log-for-v13-17

chore: change log for v13.17.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

rohitwaghchaure committed:
bumped to version 13.17.0

rohitwaghchaure committed:
Merge pull request #15424 from frappe/mergify/bp/version-13-pre-release/pr-15371

fix(ui): Rendering forms created by disabled users (backport #15371)

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

rohitwaghchaure committed:
bumped to version 13.17.1

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13
